libs/langchain-azure-dynamic-sessions/src/tools.ts

```diff
@@ -0,0 +1,260 @@
+import { promises as fs } from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { Tool } from "@langchain/core/tools";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import {
+  DefaultAzureCredential,
+  getBearerTokenProvider,
+} from "@azure/identity";
+import { v4 as uuidv4 } from "uuid";
+
+const userAgentPrefix = "langchainjs-azure-dynamic-sessions";
+
+let userAgent = "";
+async function getuserAgentSuffix(): Promise<string> {
+  try {
+    if (!userAgent) {
+      let currentDir;
+      try {
+        currentDir = __dirname;
+      } catch (e) {
+        // Workaround to make the build compatible with both ESM and CJS
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        currentDir = path.dirname(fileURLToPath(import.meta.url));
+      }
+
+      const data = await fs.readFile(
+        path.join(currentDir, "..", "package.json"),
```

Comment:
I am not sure if this will work after build etc. in all environments - have been meaning to make this a formal API. Can we remove and readd later?

Reply:
Or just hardcode as a constant for now? Since it's just going to be `0.0.1`

Reply:
Done